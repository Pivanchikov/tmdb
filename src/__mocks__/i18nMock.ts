export default jest.mock('react-i18next', () => {
  return {
    useTranslation: () => {
      return {
        t: jest.fn(value => value),
        i18n: {
          language: 'en',
          changeLanguage: jest.fn(),
        },
      }
    },
  }
})
