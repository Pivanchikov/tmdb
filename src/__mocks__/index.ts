export * from './mockGenres'
export * from './mockMovies'
export * from './storeMock'
