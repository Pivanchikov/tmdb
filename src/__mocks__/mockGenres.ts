import { Genres, GenresSelectOption } from './types'

export const enGenresMock: Genres[] = [
  { id: 28, name: 'Action' },
  { id: 12, name: 'Adventure' },
  { id: 16, name: 'Animation' },
  { id: 35, name: 'Comedy' },
  { id: 80, name: 'Crime' },
  { id: 99, name: 'Documentary' },
  { id: 18, name: 'Drama' },
  { id: 10751, name: 'Family' },
  { id: 14, name: 'Fantasy' },
  { id: 36, name: 'History' },
  { id: 27, name: 'Horror' },
  { id: 10402, name: 'Music' },
  { id: 9648, name: 'Mystery' },
  { id: 10749, name: 'Romance' },
  { id: 878, name: 'Science Fiction' },
  { id: 10770, name: 'TV Movie' },
  { id: 53, name: 'Thriller' },
  { id: 10752, name: 'War' },
  { id: 37, name: 'Western' },
]

export const enGenresSelectMock: GenresSelectOption[] = [
  { value: 28, label: 'Action' },
  { value: 12, label: 'Adventure' },
  { value: 16, label: 'Animation' },
  { value: 35, label: 'Comedy' },
  { value: 80, label: 'Crime' },
  { value: 99, label: 'Documentary' },
  { value: 18, label: 'Drama' },
  { value: 10751, label: 'Family' },
  { value: 14, label: 'Fantasy' },
  { value: 36, label: 'History' },
  { value: 27, label: 'Horror' },
  { value: 10402, label: 'Music' },
  { value: 9648, label: 'Mystery' },
  { value: 10749, label: 'Romance' },
  { value: 878, label: 'Science Fiction' },
  { value: 10770, label: 'TV Movie' },
  { value: 53, label: 'Thriller' },
  { value: 10752, label: 'War' },
  { value: 37, label: 'Western' },
]
export const ruGenresMock: Genres[] = [
  { id: 28, name: 'боевик' },
  { id: 12, name: 'приключения' },
  { id: 16, name: 'мультфильм' },
  { id: 35, name: 'комедия' },
  { id: 80, name: 'криминал' },
  { id: 99, name: 'документальный' },
  { id: 18, name: 'драма' },
  { id: 10751, name: 'семейный' },
  { id: 14, name: 'фэнтези' },
  { id: 36, name: 'история' },
  { id: 27, name: 'ужасы' },
  { id: 10402, name: 'музыка' },
  { id: 9648, name: 'детектив' },
  { id: 10749, name: 'мелодрама' },
  { id: 878, name: 'фантастика' },
  { id: 10770, name: 'телевизионный фильм' },
  { id: 53, name: 'триллер' },
  { id: 10752, name: 'военный' },
  { id: 37, name: 'вестерн' },
]

export const emptyGenre: GenresSelectOption = { value: '', label: 'Any genre' }

export const ruGenresSelectMock: GenresSelectOption[] = [
  { value: 28, label: 'Боевик' },
  { value: 12, label: 'Приключения' },
  { value: 16, label: 'Мультфильм' },
  { value: 35, label: 'Комедия' },
  { value: 80, label: 'Криминал' },
  { value: 99, label: 'Документальный' },
  { value: 18, label: 'Драма' },
  { value: 10751, label: 'Семейный' },
  { value: 14, label: 'Фэнтези' },
  { value: 36, label: 'История' },
  { value: 27, label: 'Ужасы' },
  { value: 10402, label: 'Музыка' },
  { value: 9648, label: 'Детектив' },
  { value: 10749, label: 'Мелодрама' },
  { value: 878, label: 'Фантастика' },
  { value: 10770, label: 'Телевизионный фильм' },
  { value: 53, label: 'Триллер' },
  { value: 10752, label: 'Военный' },
  { value: 37, label: 'Вестерн' },
]
