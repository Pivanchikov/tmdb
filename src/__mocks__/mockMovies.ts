/* eslint-disable no-magic-numbers */
import { TransformMovies } from './types'
import { Movie } from '../state/types'

export const movieWithAllParam: Movie = {
  adult: false,
  backdrop_path: '/las0P4Dua54XrZ73VQmGUaH1z0U.jpg',
  genre_ids: [16, 28, 18, 878],
  id: 283566,
  original_language: 'ja',
  original_title: 'シン・エヴァンゲリオン劇場版:||',
  overview:
    'In the aftermath of the Fourth Impact, stranded without their Evangelions, Shinji, Asuka, and Rei search for refuge in the desolate red remains of Tokyo-3. But the danger to the world is far from over. A new impact is looming on the horizon—one that will prove to be the true end of Evangelion.',
  popularity: 1125.051,
  poster_path: '/jDwZavHo99JtGsCyRzp4epeeBHx.jpg',
  release_date: '2021-03-08',
  title: 'Evangelion: 3.0+1.0 Thrice Upon a Time',
  video: false,
  vote_average: 8.8,
  vote_count: 194,
}

export const movieWithOutAllParam: Movie = {
  adult: false,
  backdrop_path: '/las0P4Dua54XrZ73VQmGUaH1z0U.jpg',
  genre_ids: [16, 28, 18, 878],
  id: 283566,
  original_language: 'ja',
  original_title: 'シン・エヴァンゲリオン劇場版:||',
  overview: null,
  popularity: 1125.051,
  poster_path: null,
  release_date: '2021-03-08',
  title: 'Evangelion: 3.0+1.0 Thrice Upon a Time',
  video: false,
  vote_average: 8,
  vote_count: 194,
}

export const movies: Movie[] = [
  {
    adult: false,
    backdrop_path: '/las0P4Dua54XrZ73VQmGUaH1z0U.jpg',
    genre_ids: [16, 28, 18, 878],
    id: 283566,
    original_language: 'ja',
    original_title: 'シン・エヴァンゲリオン劇場版:||',
    overview:
      'In the aftermath of the Fourth Impact, stranded without their Evangelions, Shinji, Asuka, and Rei search for refuge in the desolate red remains of Tokyo-3. But the danger to the world is far from over. A new impact is looming on the horizon—one that will prove to be the true end of Evangelion.',
    popularity: 1125.051,
    poster_path: '/jDwZavHo99JtGsCyRzp4epeeBHx.jpg',
    release_date: '2021-03-08',
    title: 'Evangelion: 3.0+1.0 Thrice Upon a Time',
    video: false,
    vote_average: 8.8,
    vote_count: 194,
  },
  {
    adult: false,
    backdrop_path: '/6MKr3KgOLmzOP6MSuZERO41Lpkt.jpg',
    genre_ids: [35, 80],
    id: 337404,
    original_language: 'en',
    original_title: 'Cruella',
    overview:
      'In 1970s London amidst the punk rock revolution, a young grifter named Estella is determined to make a name for herself with her designs. She befriends a pair of young thieves who appreciate her appetite for mischief, and together they are able to build a life for themselves on the London streets. One day, Estella’s flair for fashion catches the eye of the Baroness von Hellman, a fashion legend who is devastatingly chic and terrifyingly haute. But their relationship sets in motion a course of events and revelations that will cause Estella to embrace her wicked side and become the raucous, fashionable and revenge-bent Cruella.',
    popularity: 978.832,
    poster_path: '/wToO8opxkGwKgSfJ1JK8tGvkG6U.jpg',
    release_date: '2021-05-26',
    title: 'Cruella',
    video: false,
    vote_average: 8.4,
    vote_count: 4777,
  },
  {
    adult: false,
    backdrop_path: '/8s4h9friP6Ci3adRGahHARVd76E.jpg',
    genre_ids: [16, 35, 10751, 878],
    id: 379686,
    original_language: 'en',
    original_title: 'Space Jam: A New Legacy',
    overview:
      "When LeBron and his young son Dom are trapped in a digital space by a rogue A.I., LeBron must get them home safe by leading Bugs, Lola Bunny and the whole gang of notoriously undisciplined Looney Tunes to victory over the A.I.'s digitized champions on the court. It's Tunes versus Goons in the highest-stakes challenge of his life.",
    popularity: 2105.173,
    poster_path: '/5bFK5d3mVTAvBCXi5NPWH0tYjKl.jpg',
    release_date: '2021-07-08',
    title: 'Space Jam: A New Legacy',
    video: false,
    vote_average: 7.5,
    vote_count: 1774,
  },
  {
    adult: false,
    backdrop_path: '/xXHZeb1yhJvnSHPzZDqee0zfMb6.jpg',
    genre_ids: [28, 80, 53],
    id: 385128,
    original_language: 'en',
    original_title: 'F9',
    overview:
      "Dominic Toretto and his crew battle the most skilled assassin and high-performance driver they've ever encountered: his forsaken brother.",
    popularity: 2119.117,
    poster_path: '/bOFaAXmWWXC3Rbv4u4uM9ZSzRXP.jpg',
    release_date: '2021-05-19',
    title: 'F9',
    video: false,
    vote_average: 7.7,
    vote_count: 2804,
  },
  {
    adult: false,
    backdrop_path: '/jlGmlFOcfo8n5tURmhC7YVd4Iyy.jpg',
    genre_ids: [28, 12, 14],
    id: 436969,
    original_language: 'en',
    original_title: 'The Suicide Squad',
    overview:
      'Supervillains Harley Quinn, Bloodsport, Peacemaker and a collection of nutty cons at Belle Reve prison join the super-secret, super-shady Task Force X as they are dropped off at the remote, enemy-infused island of Corto Maltese.',
    popularity: 6294.822,
    poster_path: '/iCi4c4FvVdbaU1t8poH1gvzT6xM.jpg',
    release_date: '2021-07-28',
    title: 'The Suicide Squad',
    video: false,
    vote_average: 8.1,
    vote_count: 2435,
  },
]

export const transformMovies: TransformMovies = {
  '283566': {
    adult: false,
    backdrop_path: '/las0P4Dua54XrZ73VQmGUaH1z0U.jpg',
    genre_ids: [16, 28, 18, 878],
    id: 283566,
    original_language: 'ja',
    original_title: 'シン・エヴァンゲリオン劇場版:||',
    overview:
      'In the aftermath of the Fourth Impact, stranded without their Evangelions, Shinji, Asuka, and Rei search for refuge in the desolate red remains of Tokyo-3. But the danger to the world is far from over. A new impact is looming on the horizon—one that will prove to be the true end of Evangelion.',
    popularity: 1125.051,
    poster_path: '/jDwZavHo99JtGsCyRzp4epeeBHx.jpg',
    release_date: '2021-03-08',
    title: 'Evangelion: 3.0+1.0 Thrice Upon a Time',
    video: false,
    vote_average: 8.8,
    vote_count: 194,
  },
  '337404': {
    adult: false,
    backdrop_path: '/6MKr3KgOLmzOP6MSuZERO41Lpkt.jpg',
    genre_ids: [35, 80],
    id: 337404,
    original_language: 'en',
    original_title: 'Cruella',
    overview:
      'In 1970s London amidst the punk rock revolution, a young grifter named Estella is determined to make a name for herself with her designs. She befriends a pair of young thieves who appreciate her appetite for mischief, and together they are able to build a life for themselves on the London streets. One day, Estella’s flair for fashion catches the eye of the Baroness von Hellman, a fashion legend who is devastatingly chic and terrifyingly haute. But their relationship sets in motion a course of events and revelations that will cause Estella to embrace her wicked side and become the raucous, fashionable and revenge-bent Cruella.',
    popularity: 978.832,
    poster_path: '/wToO8opxkGwKgSfJ1JK8tGvkG6U.jpg',
    release_date: '2021-05-26',
    title: 'Cruella',
    video: false,
    vote_average: 8.4,
    vote_count: 4777,
  },
  '379686': {
    adult: false,
    backdrop_path: '/8s4h9friP6Ci3adRGahHARVd76E.jpg',
    genre_ids: [16, 35, 10751, 878],
    id: 379686,
    original_language: 'en',
    original_title: 'Space Jam: A New Legacy',
    overview:
      "When LeBron and his young son Dom are trapped in a digital space by a rogue A.I., LeBron must get them home safe by leading Bugs, Lola Bunny and the whole gang of notoriously undisciplined Looney Tunes to victory over the A.I.'s digitized champions on the court. It's Tunes versus Goons in the highest-stakes challenge of his life.",
    popularity: 2105.173,
    poster_path: '/5bFK5d3mVTAvBCXi5NPWH0tYjKl.jpg',
    release_date: '2021-07-08',
    title: 'Space Jam: A New Legacy',
    video: false,
    vote_average: 7.5,
    vote_count: 1774,
  },
  '385128': {
    adult: false,
    backdrop_path: '/xXHZeb1yhJvnSHPzZDqee0zfMb6.jpg',
    genre_ids: [28, 80, 53],
    id: 385128,
    original_language: 'en',
    original_title: 'F9',
    overview:
      "Dominic Toretto and his crew battle the most skilled assassin and high-performance driver they've ever encountered: his forsaken brother.",
    popularity: 2119.117,
    poster_path: '/bOFaAXmWWXC3Rbv4u4uM9ZSzRXP.jpg',
    release_date: '2021-05-19',
    title: 'F9',
    video: false,
    vote_average: 7.7,
    vote_count: 2804,
  },
  '436969': {
    adult: false,
    backdrop_path: '/jlGmlFOcfo8n5tURmhC7YVd4Iyy.jpg',
    genre_ids: [28, 12, 14],
    id: 436969,
    original_language: 'en',
    original_title: 'The Suicide Squad',
    overview:
      'Supervillains Harley Quinn, Bloodsport, Peacemaker and a collection of nutty cons at Belle Reve prison join the super-secret, super-shady Task Force X as they are dropped off at the remote, enemy-infused island of Corto Maltese.',
    popularity: 6294.822,
    poster_path: '/iCi4c4FvVdbaU1t8poH1gvzT6xM.jpg',
    release_date: '2021-07-28',
    title: 'The Suicide Squad',
    video: false,
    vote_average: 8.1,
    vote_count: 2435,
  },
}

export const mockMovies = [[movies, transformMovies]]
