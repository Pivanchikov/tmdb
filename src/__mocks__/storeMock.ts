import { MoviesState } from '../state/types'

export const initialState: MoviesState = {
  items: [],
  byId: {},
  genres: [],
  selectedMovie: null,
  isLoading: false,
  isFoundMovie: false,
  popularMovies: [],
  trendingMovies: [],
  nowPlayingMovies: [],
  upcomingMovies: [],
  topRatedMovies: [],
}
