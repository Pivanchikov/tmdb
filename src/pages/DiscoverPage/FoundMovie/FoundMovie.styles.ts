/* eslint-disable @typescript-eslint/prefer-as-const */
export const styles = theme => ({
  container: {
    width: 720,
    margin: '30px auto',
    padding: 20,
    display: 'flex',
    boxShadow: `0 0 0.4em ${theme.palette.primary.main}`,
  },
  movieInfo: {
    paddingLeft: 20,
    width: '100%',
  },
  poster: {
    transition: '0.5s',
    border: `1px solid ${theme.palette.primary.main}`,
    backgroundColor: '#eeeeee',
    '&:hover': {
      transform: 'scale(1.05)',
      cursor: 'pointer',
    },
  },
  header: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'baseline',
  },
  title: {
    color: theme.palette.primary.main,
    '&:hover': {
      textDecoration: 'underline',
      cursor: 'pointer',
    },
  },
  vote: {
    backgroundColor: theme.palette.primary.main,
    width: 60,
    textAlign: 'center' as 'center',
    color: theme.palette.secondary.main,
  },
  date: {
    color: theme.palette.primary.main,
    fontSize: '0.8rem',
  },
  overview: {
    color: theme.palette.primary.main,
    textAlign: 'justify' as 'justify',
    marginTop: '1rem',
  },
  footer: {
    display: 'flex',
    justifyContent: 'flex-end',
  },
  button: {
    backgroundColor: theme.palette.primary.main,
    margin: '0 5px',
    fontSize: '0.7rem',
    '&:hover': {
      backgroundColor: theme.palette.primary.light,
    },
  },
  movieNotFound: {
    color: theme.palette.primary.main,
    margin: '0 auto',
  },

  '@media screen and (max-width: 959px)': {
    container: {
      width: 360,
      flexDirection: 'column' as 'column',
      textAlign: 'center' as 'center',
    },
    movieInfo: {
      padding: '10px 0',
    },
    header: {
      padding: '10px 0',
      maxWidth: 400,
      justifyContent: 'space-around',
      margin: '0 auto',
    },
  },
  '@media screen and (max-width: 370px)': {
    container: {
      width: 320,
    },
  },
})
