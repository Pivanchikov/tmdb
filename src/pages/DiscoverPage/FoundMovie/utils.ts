import noImage from '../../../assets/img/noImage.svg'
import { Movie } from '../../../state/types'

const voteLength = 1

export const choicePosterPath = path => (path ? `https://image.tmdb.org/t/p/w200${path}` : noImage)
export const formatDate = date => new Date(date).getFullYear()
export const formatVote = (vote: number) => (Number.isInteger(vote) ? vote.toFixed(voteLength) : vote)

export const getVote = (movie: Movie | null) => (movie ? formatVote(movie.vote_average) : '')
export const getYear = (movie: Movie | null) => (movie ? formatDate(movie.release_date) : '')
export const getPosterPath = (movie: Movie | null) => (movie ? choicePosterPath(movie.poster_path) : '')
