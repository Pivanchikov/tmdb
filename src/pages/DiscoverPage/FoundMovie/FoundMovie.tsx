import { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { makeStyles, Typography } from '@material-ui/core'
import { Redirect } from 'react-router-dom'
import { useTranslation } from 'react-i18next'

import { getSelectedMovie, setMovieFound } from '../../../state'
import { AverageVote } from '../../../components'
import { styles } from './FoundMovie.styles'
import { getPosterPath, getYear, getVote } from './utils'

const useStyles = makeStyles(styles)

export const FoundMovie = () => {
  const dispatch = useDispatch()
  const classes = useStyles()
  const { t } = useTranslation()
  const movie = useSelector(getSelectedMovie)

  const [redirect, setRedirect] = useState(false)

  const redirectToMovie = async () => {
    await setRedirect(true)
    dispatch(setMovieFound(false))
  }

  const vote = getVote(movie)
  const year = getYear(movie)
  const overview = (movie && movie.overview) || t('overview')

  const posterPath = getPosterPath(movie)
  const redirectPath = movie && `/movie/${movie.id}-${movie.title}`

  return (
    <>
      {redirect && <Redirect to={redirectPath} />}
      <section className={classes.container}>
        {movie ? (
          <>
            <div className="poster">
              <img
                src={posterPath}
                alt="poster"
                width="200px"
                height="300px"
                className={classes.poster}
                onClick={redirectToMovie}
                data-testid={'poster'}
              />
            </div>
            <div className={classes.movieInfo}>
              <header className={classes.header}>
                <Typography
                  variant="h5"
                  className={classes.title}
                  onClick={redirectToMovie}
                  data-testid={'title'}
                >
                  {movie.title}
                </Typography>
                <AverageVote vote={vote} />
              </header>
              <Typography className={classes.date} data-testid={'date'}>
                {year}
              </Typography>
              <Typography className={classes.overview} data-testid={'overview'}>
                {overview}
              </Typography>
            </div>
          </>
        ) : (
          <Typography variant="h5" className={classes.movieNotFound}>
            Movie not found
          </Typography>
        )}
      </section>
    </>
  )
}
