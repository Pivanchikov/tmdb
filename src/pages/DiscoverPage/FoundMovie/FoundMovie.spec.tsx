import { render as testingLibraryRender, fireEvent } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'

import { movieWithAllParam, movieWithOutAllParam } from '../../../__mocks__'
import { FoundMovie } from './FoundMovie'

const mockStore = configureStore([])

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: jest.fn(value => value),
    i18n: {
      language: 'en',
      changeLanguage: jest.fn(),
    },
  }),
}))

describe('Found Movie component', () => {
  let store

  const render = () =>
    testingLibraryRender(
      <Router history={createMemoryHistory()}>
        <Provider store={store}>
          <FoundMovie />
        </Provider>
      </Router>
    )
  it('render with all keys', () => {
    store = mockStore({
      movies: { items: [], byId: {}, selectedMovie: movieWithAllParam },
    })

    const { getByText } = render()

    expect(getByText(movieWithAllParam.vote_average)).toBeInTheDocument
    expect(getByText(movieWithAllParam.title)).toBeInTheDocument
  })

  it('render without all keys', () => {
    store = mockStore({
      movies: { items: [], byId: {}, selectedMovie: movieWithOutAllParam },
    })

    const { getByText } = render()

    expect(getByText('8.0')).toBeInTheDocument
    expect(getByText(movieWithOutAllParam.title)).toBeInTheDocument
    expect(getByText('overview')).toBeInTheDocument
  })

  it('click on label for redirect', () => {
    store = mockStore({
      movies: { items: [], byId: {}, selectedMovie: movieWithAllParam },
    })

    const { getByTestId } = render()

    fireEvent.click(getByTestId('title'))
  })

  it('matches snapshot', () => {
    const { asFragment } = render()

    expect(asFragment()).toMatchSnapshot()
  })
})
