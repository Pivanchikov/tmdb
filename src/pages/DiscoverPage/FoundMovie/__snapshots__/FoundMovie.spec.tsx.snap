// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Found Movie component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="makeStyles-container-43"
  >
    <div
      class="poster"
    >
      <img
        alt="poster"
        class="makeStyles-poster-45"
        data-testid="poster"
        height="300px"
        src="https://image.tmdb.org/t/p/w200/jDwZavHo99JtGsCyRzp4epeeBHx.jpg"
        width="200px"
      />
    </div>
    <div
      class="makeStyles-movieInfo-44"
    >
      <header
        class="makeStyles-header-46"
      >
        <h5
          class="MuiTypography-root makeStyles-title-47 MuiTypography-h5"
          data-testid="title"
        >
          Evangelion: 3.0+1.0 Thrice Upon a Time
        </h5>
        <p
          class="MuiTypography-root makeStyles-green-54 MuiTypography-body1"
          data-testid="vote"
        >
          8.8
        </p>
      </header>
      <p
        class="MuiTypography-root makeStyles-date-49 MuiTypography-body1"
        data-testid="date"
      >
        2021
      </p>
      <p
        class="MuiTypography-root makeStyles-overview-50 MuiTypography-body1"
        data-testid="overview"
      >
        In the aftermath of the Fourth Impact, stranded without their Evangelions, Shinji, Asuka, and Rei search for refuge in the desolate red remains of Tokyo-3. But the danger to the world is far from over. A new impact is looming on the horizonâ€”one that will prove to be the true end of Evangelion.
      </p>
    </div>
  </section>
</DocumentFragment>
`;
